jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/Validator.java

```diff
@@ -27,35 +28,116 @@
    *
    * @param value the string in question
    * @param propertyName the equivalent 'yaml' property name
    * @throws NullPointerException if {@code value} is null
    * @throws IllegalArgumentException if {@code value} is empty
    */
-  public static void checkNotEmpty(@Nullable Collection<?> value, String propertyName) {
+  public static void checkNotNullAndNotEmpty(@Nullable Collection<?> value, String propertyName) {
     Preconditions.checkNotNull(value, "Property '" + propertyName + "' cannot be null");
     Preconditions.checkArgument(
         !value.isEmpty(), "Property '" + propertyName + "' cannot be an empty collection");
   }
 
   /**
-   * Checks if string is what is expected.
+   * Check if a collection is either null, empty or contains only non-null, non-empty values.
+   *
+   * @param values the collection in question
+   * @param propertyName the equivalent 'yaml' property name
+   * @throws IllegalArgumentException if {@code value} is empty or only whitespace
+   */
+  public static void checkNonNullNonEmptyEntriesIfExists(
+      @Nullable Collection<String> values, String propertyName) {
+    if (values == null || values.isEmpty()) {
+      // pass
+      return;
+    }
+    for (String value : values) {
+      Preconditions.checkNotNull(
+          value, "Property '" + propertyName + "' cannot contain null entries");
+      Preconditions.checkArgument(
+          !value.trim().isEmpty(), "Property '" + propertyName + "' cannot contain empty entries");
+    }
+  }
+
+  /**
+   * Check if a map is either null, empty or contains only non-null, non-empty keys and values.
+   *
+   * @param values the collection in question
+   * @param propertyName the equivalent 'yaml' property name
+   * @throws IllegalArgumentException if {@code value} is empty or only whitespace
+   */
+  public static void checkNonNullNonEmptyEntriesIfExists(
+      @Nullable Map<String, String> values, String propertyName) {
+    if (values == null || values.isEmpty()) {
+      // pass
+      return;
+    }
+    for (String key : values.keySet()) {
+      Preconditions.checkNotNull(key, "Property '" + propertyName + "' cannot contain null keys");
+      Preconditions.checkArgument(
+          !key.trim().isEmpty(), "Property '" + propertyName + "' cannot contain empty keys");
+      String value = values.get(key);
+      Preconditions.checkNotNull(
+          value, "Property '" + propertyName + "' cannot contain null values");
+      Preconditions.checkArgument(
+          !value.trim().isEmpty(), "Property '" + propertyName + "' cannot contain empty values");
+    }
+  }
+
+  /**
+   * Check if a collection is either null, empty or contains only non-null values.
+   *
+   * @param values the collection in question
+   * @param propertyName the equivalent 'yaml' property name
+   * @throws IllegalArgumentException if {@code value} is empty or only whitespace
+   */
+  public static void checkNonNullEntriesIfExists(
```

Comment:
`IfExists` sounds redundant to me when covering an empty collection. It doesn't cover null input though. I think `checkNullOrNonNullEntries()` coveys the intention more clearly, because it's difficult to know what "IfExists" means (== not null and collection is not empty") without reading the Javadoc. I initially thought "IfExists" means "if an entry exists at all." WDYT?

Reply:
yeah that makes sense to me, I was having trouble naming this.